platform-common-pdf/src/main/java/com/softicar/platform/common/pdf/PdfRenderer.java

```diff
@@ -73,36 +99,89 @@ public PdfRenderer setImageType(ImageType imageType) {
 	 */
 	public List<BufferedImage> render(InputStream inputStream) {
 
-		try (var document = PDDocument.load(inputStream)) {
-			return render(document);
+		byte[] bytes = new ByteBuffer(() -> inputStream).getBytes();
+
+		try (var document = PDDocument.load(bytes)) {
+			var workers = new ArrayList<PageRenderingWorker>();
+			for (var page = 0; page < document.getNumberOfPages(); page++) {
+				workers.add(new PageRenderingWorker(dpi, imageType, bytes, page));
+			}
+
+			var threadRunner = new LimitedThreadRunner<PageRenderingWorker>(maxRenderingThreads);
+			workers.forEach(threadRunner::addRunnable);
+			do {
+				threadRunner.startThreads();
+				Sleep.sleep(Duration.ofMillis(10));
+			} while (!threadRunner.isFinished());
+
+			return workers//
+				.stream()
+				.map(PageRenderingWorker::getImage)
+				.collect(Collectors.toList());
 		} catch (IOException exception) {
 			throw new SofticarIOException(exception, "Failed to render PDF.");
 		}
 	}
 
-	/**
-	 * Renders the pages of the {@link PDDocument} into one or more
-	 * {@link BufferedImage} instances.
-	 * <p>
-	 * The given {@link PDDocument} will <b>not</b> be closed by this method.
-	 *
-	 * @param document
-	 *            the {@link PDDocument} to render; needs to be closed by the
-	 *            caller (never <i>null</i>)
-	 * @return list of rendered images; one per page (never <i>null</i>)
-	 */
-	public List<BufferedImage> render(PDDocument document) {
+	private static class PageRenderingWorker implements Runnable {
```

Comment:
since you already added Javadoc, I suggest to extract this class as public `PdfPageRenderer` or `PdfSinglePageRenderer`, or similar

Reply:
Extracted to `PdfSinglePageRenderer`. Remains package-visible though. Also simplified the code.